LoopKitUI/Extensions/Environment+AppName.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  Environment+AppName.swift
+//  LoopUI
+//
+//  Created by Rick Pasetto on 7/1/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import SwiftUI
+
+private struct AppNameKey: EnvironmentKey {
+    static let defaultValue = Bundle.main.object(forInfoDictionaryKey: "CFBundleDisplayName") as! String
```

Comment:
Using Environment to share this value across swiftui views is perfect, but I don't think the LoopKitUI framework should assume it can reach up into the App's bundle and grab the app name from there. Can we make this configurable instead, so that users of these views can set it to what they want?

Reply:
In a sense, they do.  "Users" in this case is any app that uses `LoopKit` and sets their `CFBundleDisplayName` in the project plist.  No?

Reply:
Could be; could be some other use. Frameworks can be used by other frameworks, with other needs.  Reusability would dictate that the class not make assumptions about being in an executable that has a bundle with a "CFBundleDisplayName" key in it, but just provide a straightforward way for it to be set. This is by no means a hill I want to die on, and will be ok with it if you want to go this way, as I said, but I do think it's cleaner and less tightly coupled to not make those assumptions.

Reply:
@ps2 It's looking like we'll need some mechanism for configuration within LoopKit to be able to remove "Tidepool" from some other strings for the DIY merge. What do you think is a good option here? It might not be pretty, but some struct with public static variables that can be set from the top-level is one option. (We could at least enforce that they're set only once, to make them less like global mutable state.)

Reply:
You're asking about https://tidepool.atlassian.net/browse/LOOP-1422, @mpangburn ?  Yeah, that's another aspect of this.  Does `LoopKit` (or `LoopKitUI`) have a static initializer?  That'd be one way to do it.

Reply:
@rickpasetto That's a related ticket, sure. We also have copy like "The value you have entered is higher than Tidepool generally recommends", where the string "Tidepool" will have to be updated to be pulled down from a higher level, or made settable. Off the top of my head, I can think of two paths:
1. Use the app's main bundle with a key defined by LoopKitUI, and use the environment to grab it (i.e. analagous to what this PR does with the app name, but requiring a new bundle key). As @ps2 mentions the drawback here is that the top-level app has to know about this framework requirement; I think Loop and LoopKit are tightly enough coupled that this wouldn't be a problem for our use case, though it would mean that a theoretical new framework couldn't abstract over LoopKit without its top-level app being aware of it.
2. Define some global variable in LoopKitUI, and in `applicationDidFinishLaunching` in Loop assign the appropriate values. This "feels" messier at the code level because it's global state, but is technically a looser coupling with the app itself.

I mention this because if we think we'll go path 2 (or some other path I haven't considered), then the way `appName` is passed down should also use that mechanism.

Reply:
I'm not sure just a blanket substitution will work here. Note that in "Tidepool generally recommends" it is *not* "Tidepool Loop generally recommends" (as in the name of the app). 

The better bet may be to go for a "Derived" version of the strings files.

Reply:
I don't think global vars or a "system" approach is needed here. I get that we're all programmers and if we detect a sniff of something repeating, we want to systemize it.  But I would like to challenge that and suggest any new places where we need configurability in a view to just expose a member or initializer param to the view where the client can control it. Systems are great until assumptions break and then the destruction is widespread.  Let's implement the needed configurability in the simplest way possible, and then later systemize it as a refactoring if we really feel that's needed.

Reply:
@darinkrauss's comment that some cases are specific is a very good example of this: even for something that seems as straightforward as "app name" might not be as straightforward as we initially think. What if a user of these views wants to use an abbreviation of the app name in some contexts, or maybe for DIY Loop we want to keep the name under the icon "Loop", but we decide that the view should read "DIY Loop" to be more explicit.

Reply:
Exposing a simple member on the view, or an constructor parameter leaves all those concerns with the client, where it should be.

Reply:
@ps2 @darinkrauss @mpangburn These are all good points.  However, I wonder if we're not over-thinking this.  I agree that global substitution will not work everywhere.  That's fine.  However, simply put, sometimes you do really just want the app name.  The name of this variable is clear: it isn't the name of the company, or the generic name "Loop"... it is actually the name of the app, as specified by iOS and defined in the Bundle.  If you don't want to use it, don't use it.  What I think Michael and I are trying to clean up is the "injection infection" of passing around `appName` everywhere.  This is one of those few places where I think a "global constant" is merited.  _Where_ it is used as a substitute is beyond the scope of this PR.

Reply:
If this was in Loop itself, there would be no issue. LoopKit has a higher bar by having a goal of reusability, which I think in this case means not making assumptions about things outside of LoopKit.  I think this particular assumption (that LoopKit is being used by an app that will provide the bundle and the key in it) is probably going to hold true, so I don't think this will break anything, and I will not block it, but it still feels dirty.

All I'm asking is that you remove the reachback, so the API call in Loop looks like this:

`SettingsView(viewModel: viewModel).environment(\.appName, Bundle.main.bundleDisplayName)`

The environment extension can live in LoopKit, and all the views can still use it; it will just need to be set at top level views, and all subviews can use it. I think the Environment is carrying the weight of managing "injection infection".

Reply:
One problem, @ps2 : these screens are also used by _TidepoolServiceKit_.  So, how will _TidepoolServiceKit_ get the app name?

Reply:
Yeah, that needs to be solved as well. Plugins are different in the use cases they support; while LoopKit is a general purpose ToolKit, Loop plugins are designed to work specifically with Loop and only with Loop, and I think making the reach into the app bundle is a safer assumption there. If we wanted to keep that assumption out, we could add it to the service delegate as well, which is another option.  In either case, I think the solution here in LoopKit to just expose via the Environment is still fine, and allows clients to fulfill the requirement however they want.

Reply:
@ps2 ok.  For now, I'll let the plugin pull from Bundle.main.